src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1141 @@
+      {
+        l1Token: params.erc20L1Address,
+        l1l2Router: this.l2Network.tokenBridge.l1GatewayRouter,
+        l2l3Router: this.l3Network.tokenBridge.l1GatewayRouter,
+        to: params.to || (await l1Signer.getAddress()),
+        amount: params.amount,
+        gasParams,
+      },
+    ])
+
+    const adjustedL1GasPrice = this._percentIncrease(
+      params.overrides?.l1GasPrice?.base ||
+        (await l1Signer.provider!.getGasPrice()),
+      params.overrides?.l1GasPrice?.percentIncrease ||
+        this.defaultGasPricePercentIncrease
+    )
+
+    const calculatedGasCosts = await teleporter.calculateRetryableGasCosts(
+      this.l2Network.ethBridge.inbox,
+      adjustedL1GasPrice,
+      gasParams
+    )
+
+    return {
+      to: this.teleporterAddresses.l1Teleporter,
+      data: calldata,
+      value: calculatedGasCosts.total,
+    }
+  }
+
+  /**
+   * Deposit tokens to L3 through the L1Teleporter contract.
+   */
+  public async deposit(
+    params: Erc20DepositRequestParams,
+    l1Signer: Signer,
+    l2Provider: Provider,
+    l3Provider: Provider
+  ): Promise<L1ContractCallTransaction> {
+    const txRequest = await this.getDepositRequest(
+      params,
+      l1Signer,
+      l2Provider,
+      l3Provider
+    )
+
+    return L1TransactionReceipt.monkeyPatchContractCallWait(
+      await l1Signer.sendTransaction(txRequest)
+    )
+  }
+
+  /**
+   * Get the status of a deposit given an L1 tx receipt. See `Erc20DepositStatus` interface for more info on the return type.
+   *
+   * Note: This function does not verify that the tx is actually a deposit tx.
+   */
+  public async getDepositStatus(
+    depositTxReceipt: L1ContractCallTransactionReceipt,
+    l2Provider: JsonRpcProvider,
+    l3Provider: Provider
+  ): Promise<Erc20DepositStatus> {
+    const baseStatus = await this._getDepositStatus(
+      depositTxReceipt,
+      l2Provider,
+      l3Provider
+    )
+
+    const secondStepRetryableRedeem = await (
+      await depositTxReceipt.getL1ToL2Messages(l2Provider)
+    )[1].getSuccessfulRedeem()
+
+    return {
+      ...baseStatus,
+      retryableL2ForwarderCall: secondStepRetryableRedeem,
+    }
+  }
+}
+
+/**
+ * Bridge ERC20 tokens from L1 to L3 directly through the token bridge, using a relayer on L2 to call the L2Forwarder contract.
+ */
+export class RelayedErc20L1L3Bridger extends BaseErc20L1L3Bridger {
+  /**
+   * The default percent increase for the relayer payment
+   */
+  public readonly defaultRelayerPaymentPercentIncrease: BigNumber =
+    BigNumber.from(30)
+
+  /**
+   * Get a tx request to approve tokens for teleportation. Will approve the token's L1 gateway to spend the provided amount.
+   */
+  public async getApproveTokenRequest(
+    params: TokenApproveParams,
+    l1Provider: Provider
+  ): Promise<Required<Pick<TransactionRequest, 'to' | 'data'>>> {
+    return new Erc20Bridger(this.l2Network).getApproveTokenRequest({
```

Comment:
it seems weird to use the ERC20Bridger class here just for the approval, maybe instead have BaseErc20L1L3Bridger.approveToken with an optional param to allow specifying the approval address